src/postprocessing/least_squares.jl

```diff
@@ -156,37 +158,53 @@ See also [`BHC`](@ref).
 - `x_0`: First guess for the initial conditions.
 - `niters`: Number of iterations.
 """
-function diffcorr(res, w, x0, niters=5)
+function diffcorr(res::Vector{TaylorN{T}}, w::Vector{T}, x0::Vector{T}, niters::Int = 5) where {T <: Real}
     # Have as many residuals as weights
     @assert length(res) == length(w)
-    # Number of observations
-    nobs = length(res)
     # Degrees of freedom
     npar = length(x0)
     # Design matrix B, H array and normal matrix C
     B_mat, H_mat, C_mat = BHC(res, w, npar)
     # D matrix: transpose(B) * W * ξ
     D_mat = B_mat' * (w .* res)
     # ξTH_mat = ξTH(w, res, H_mat, npar)
+    # Vector of x 
+    x = Matrix{T}(undef, niters + 1, npar) 
     # First guess
-    x_new = x0
+    x[1, :] = x0
+    # Vector of errors 
+    error = Vector{T}(undef, niters + 1)
+    # Error of first guess 
+    error[1] = T(Inf)
     # Iteration
     for i in 1:niters
         # D matrix evaluated in x_new
-        D = D_mat(x_new)
+        D = D_mat(x[i, :])
```

Comment:
I guess this is better (performance) to handle as a view, e.g.
```julia
xi = view(x, i, :)
D = D_mat(xi)
```

(But don't take my word for granted...)

If you include my previous comment, adapt the order of the indices.

Reply:
Replace below `x[i, :]` by `xi`...

Reply:
I've tested it and it turns out that using views do not improve performance significantly. Also `D_mat(xi)` throws an error for `xi = view(x, i, :)`. Should I leave it as it is?

Reply:
If `view` doesn't help, let's leave it as it is; marking this comment as solved.